.github/workflows/ci.yml

```diff
@@ -63,3 +63,41 @@ jobs:
         run: pip install -e '.[dev]'
       - name: Run tests
         run: pytest -v -rs --log-level=DEBUG tests
+
+  release:
+    needs:
+      - check-copyright
+      - lint
+      - tests
+    if: github.ref_name == 'main'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+      - name: Install package
+        run: pip install -e .
+      - name: Figure out current version
+        run: python -c 'import benchalerts; print(f"VERSION={benchalerts.__version__}")' >> $GITHUB_ENV
+      - name: Create or update the git tag with this version
+        uses: actions/github-script@v6
+        with:
+          script: |
+            try {
+              await github.rest.git.createRef({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                ref: 'refs/tags/${{ env.VERSION }}',
+                sha: context.sha
+              });
+            } catch (error) {
+              console.log(error);
+              console.log('Error creating a ref; will try to update it instead');
+              await github.rest.git.updateRef({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                ref: 'tags/${{ env.VERSION }}',
+                sha: context.sha
+              });
```

Comment:
Am I following that this will actually be changing which sha has the 0.1.0 on each commit (until we have a 0.1.1, and then only the last 0.1.0 would have that?).

Reply:
Yes, that's correct. I'm not particularly tied to this strategy though. We could alternatively only tag the first SHA of a given version.

Reply:
Reason I chose this strategy is there could be PRs that change stuff like docstrings, which may not constitute a patch version. But maybe that's against the idea of semver.

Reply:
What if we tagged each with a `0.1.0-short-sha` and then the "release" gets the real one? Though IIUC one could get the specific sha with pip already?

Reply:
Mostly: it feels bad to me to change the commit that a tag points to pretty consistently.

Reply:
From a utility standpoint, I'd rather it just be the semver; otherwise I'd just go install from the commit anyway.

Reply:
Yeah, someone could always get a specific SHA with pip. The "real" way to do it is probably

- tag the first SHA with a given `MAJOR.MINOR.PATCH`
- have a running `MAJOR.MINOR` tag that gets updated with each new patch
- have a running `MAJOR` tag that gets updated with each new major

But at that level of complexity we might as well just start uploading to PyPI.

I agree that changing the tag over and over could cause problems. I already had to turn off "always fetch tags" in my VS code settings because it started running into the git "error: fetching would clobber an existing tag." So maybe we switch to just tagging the first SHA with `MAJOR.MINOR.PATCH` and if someone makes any sort of update to release they have to bump the respective version. We can revisit this in the future, especially when we add PyPI infrastructure.

Reply:
Okay, I changed the code to only tag the first SHA for a given version.